src/main/java/vazkii/botania/common/core/helper/MathHelper.java

```diff
@@ -50,4 +51,12 @@ public static int multiplyColor(int c1, int c2) {
 		return c1 & ~0xFFFFFF | r << 16 | g << 8 | b;
 	}
 
+	public static int distSqr(BlockPos a, BlockPos b) {
+		//Vec3i#distSqr, while convenient, offsets the second argument by (0.5, 0.5, 0.5).
+		//It also returns a double, which is unneeded; the squared distance between two integer coordinates is an integer.
+		int dx = a.getX() - b.getX();
+		int dy = a.getY() - b.getY();
+		int dz = a.getZ() - b.getZ();
+		return dx * dx + dy * dy + dz * dz;
```

Comment:
I can't help but worry that some sort of shenanigans is possible if you try to somehow bind something near the world border to spawn chunks. Please test.

Reply:
Shenanigans are possible lol
![image](https://user-images.githubusercontent.com/2068021/135775717-d15444b5-d22a-446a-bfae-3a48c03fbfc5.png)
i was able to bind a flower at 47000, 0 to a pool at 0, 0 because of the overflow

Reply:
Fixded by using longs which overflow at 3 billion, way bigger than a mc world. (I guess i could have used doubles like vanilla)

Reply:
I don't really get why we're using longs here instead of double. Doesn't that lose precision?

Reply:
Oh you're multiplying integer inputs I'm stupid.